src/fpm_compiler.f90

```diff
@@ -398,6 +399,18 @@ subroutine set_preprocessor_flags (flags)
         return
     end if
 
+    !> Modify the flag_cpp_preprocessor on the basis of the compiler.
+    select case(id)
+    case default
+        flag_cpp_preprocessor = ""
+    case(id_caf, id_gcc, id_f95, id_nvhpc)
+        flag_cpp_preprocessor = "-cpp"
+    case(id_intel_classic_nix, id_intel_classic_mac, id_intel_classic_windows, id_intel_llvm_nix, id_intel_llvm_windows, id_nag)
+        flag_cpp_preprocessor = "-fpp"
```

Comment:
Thanks for your great work @arteevraina. `id_intel_classic_windows` and `id_intel_llvm_windows` should be `/fpp`, and the preprocess option of Intel's Fortran compiler on Windows OS is `/fpp`, not `-fpp`.

PS: It is worth noting that on Linux and Mac systems, most compilers enable preprocessing by default for `.F90` files, but on Windows systems, Intel's compilers do not enable preprocessing **by default** for `.F90` files.

Reply:
Thanks for reviewing @zoziha. I think if we make changes as per the your comments. I think the final logic should look like this one :

```
    !> Modify the flag_cpp_preprocessor on the basis of the compiler.
    select case(id)
    case default
        flag_cpp_preprocessor = ""
    case(id_caf, id_gcc, id_f95, id_nvhpc)
        flag_cpp_preprocessor = "-cpp"
    case(id_intel_classic_windows, id_intel_llvm_windows)
        flag_cpp_preprocessor = "/fpp"
    case(id_intel_classic_nix, id_intel_classic_mac, id_intel_llvm_nix, id_nag)
        flag_cpp_preprocessor = "-fpp"
    case(id_lfortran)
        flag_cpp_preprocessor = "--cpp"
    end select
```
Can you confirm this ?